### PR TITLE
feat: add update_balance to vm::testing::Contract

### PR DESCRIPTION
**Pull request by loloicci**
# Description
Add `update_balance` to `vm::testing::Contract`.
It will be used by cosmwasm-simulator.

## Types of changes

- [ ] Bug fix (changes which fixes an issue)
- [x] New feature (changes which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] ETC (build, ci, docs, perf, refactor, style, test)

## Checklist

- [x] I followed the [contributing guidelines](https://github.com/line/cosmwasm/blob/main/CONTRIBUTING.md).
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] The PR title and commits are followed [conventional commit form](https://www.conventionalcommits.org/en/v1.0.0).

### CODE REVIEW

File: packages/vm/src/testing/contract.rs [hunk 1]
```diff
@@ -95,15 +99,30 @@ impl<'a> Contract<'a> {
     pub fn change_options(&mut self, options: MockInstanceOptions<'a>) {
         self.options = options
     }
+
+    /// update balance in backend querier. it does not change the options
+    pub fn update_balance<T: Into<String>>(&mut self, addr: T, balance: Vec<Coin>) -> TestingResult<Option<Vec<Coin>>> {
+        let mut backend = self
+            .backend
+            .take()
```
Comment (brew0722) [inline packages/vm/src/testing/contract.rs:107]: Do we have to 'take'? how about just use to 'match' and not assigning the backend again?

File: packages/vm/src/testing/contract.rs [hunk 2]
```diff
@@ -95,15 +99,30 @@ impl<'a> Contract<'a> {
...
+            .ok_or_else(|| TestingError::ContractError(ERR_UPDATE_BALANCE_OF_INSTANCE.to_string()))?;
+
+        // update balances of querier
+        let res = backend.querier.update_balance(addr, balance);
+        self.backend = Some(backend);
```
Comment (brew0722) [inline packages/vm/src/testing/contract.rs:112]: please see the above opinion. I think if we not use the `take`,  we don't need this line.

File: packages/vm/src/testing/contract.rs (hunk 1 above)
Comment (loloicci) [reply packages/vm/src/testing/contract.rs]: Thanks! reflected!

File: packages/vm/src/testing/contract.rs (hunk 2 above)
Comment (loloicci) [reply packages/vm/src/testing/contract.rs]: ditto.

**Review by brew0722 — APPROVED** [review]

**Comment by loloicci** [comment]
I close this because #181 makes this change out of the scope of `Contract`.

**Pull request closed without merging**